### PR TITLE
Feat/collapsible if else 

### PR DESCRIPTION
**Pull request by lindsaymoralesb**
## Issue
#8 

## Description 

- Lint implementation
- Fix implementation
- Tests added

### CODE REVIEW

File: crates/cairo-lint-core/src/lints/collapsible_if_else.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                true
+            } else {
+                false
+            }
+        } else {
+            false
+        }
+    } else {
+        false
+    }
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/lints/collapsible_if_else.rs:28]: too many indents here group the ifs together like that
```suggestion
    if let Some(first_statement) = statements.first() && let Statement::Expr(statement_expr) = first_statement && let Expr::If(_) = first_statement_expr = statement_expr.expr(db) {
    true
    } else {
        false
    }
```

File: crates/cairo-lint-core/src/fix.rs [hunk 2]
```diff
@@ -206,6 +210,237 @@ impl Fixer {
...
+    
+    pub fn fix_collapsible_if_else(&self, db: &dyn SyntaxGroup, node: SyntaxNode) -> String {
+        // Call the transformation function to handle collapsible if-else
+        let fixed_text = self.transform_if_else(node.get_text(db));
+
+        fixed_text
+    }
+
+    // Transforms text to replace "else { if" pattern with "else if"
+    fn transform_if_else(&self, text: String) -> String {
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/fix.rs:236]: looks extremely complex, try working with the tree directly not the text. you can do it by replacing the `else` node with the `if` node text and prefix it with `else`

**Review by 0xLucqs — CHANGES_REQUESTED** [review]

File: crates/cairo-lint-core/src/lints/collapsible_if_else.rs (hunk 1 above)
Comment (lindsaymoralesb) [reply crates/cairo-lint-core/src/lints/collapsible_if_else.rs]: That suggestion gave some issues, but I improved the code in another way 😊

File: crates/cairo-lint-core/src/fix.rs (hunk 2 above)
Comment (lindsaymoralesb) [reply crates/cairo-lint-core/src/fix.rs:236]: got it, working on it!

**Comment by lindsaymoralesb** [comment]
hey @0xLucqs comments addressed!

1. For the nested ifs change on the linter your suggestion was generating some compilation issues, so I fixed it in another way.
2. For the complex fix, I changed the approach I was using (with the text) and instead started using the tree. This is working good, I just noticed that there is a small indentation issue with the fix on the else if clause. Do you have an idea of how can I approach that small issue?

**Comment by 0xLucqs** [comment]
let's ignore complex indentation problems it's a nightmare

**Comment by lindsaymoralesb** [comment]
> let's ignore complex indentation problems it's a nightmare

Got it! Then the PR is ready for review.

**Review by 0xLucqs — DISMISSED** [review]

**Review by 0xLucqs — APPROVED** [review]

**Comment by lindsaymoralesb** [comment]
@0xLucqs is there any action needed from my side to make the ci/cd run correctly?

**Comment by 0xLucqs** [comment]
Yep you need to run `cargo fmt`

**Comment by lindsaymoralesb** [comment]
> Yep you need to run `cargo fmt`

done! @0xLucqs

**Review by 0xLucqs — APPROVED** [review]

**Comment by 0xLucqs** [comment]
@all-contributors please add @lindsaymoralesb for code and test

**Pull request merged**